src/LogicTasks/Syntax/SubTreeSet.hs

```diff
@@ -124,37 +133,42 @@ partialGrade' SubTreeInst{..} fs
 
 
 completeGrade
-  :: (OutputCapable m, MonadIO m)
+  :: (OutputCapable m, MonadIO m, Alternative m)
   => FilePath
   -> SubTreeInst
   -> Delayed [FormulaAnswer]
-  -> LangM m
+  -> Rated m
 completeGrade path inst = completeGrade' path inst `withDelayedSucceeding` parser
 
 completeGrade'
-  :: (OutputCapable m, MonadIO m)
+  :: (OutputCapable m, MonadIO m, Alternative m)
   => FilePath
   -> SubTreeInst
   -> [FormulaAnswer]
-  -> LangM m
-completeGrade' path SubTreeInst{..} sol = refuseIfWrong $ do
-  unless partOfSolution $ do
+  -> Rated m
+completeGrade' path SubTreeInst{..} sol = reRefuse
+  (extendedMultipleChoice
+    (MinimumThreshold 0)
+    (Punishment 0)
+    (TargetedCorrect (fromIntegral minInputTrees))
+    IndefiniteArticle
+    what
+    Nothing
+    solution
+    (map show sol))
+  $ when showSolution $ indent $ do
```

Comment:
Won't this now show the solution (unnecessarily) even when the student submission was correct?

Reply:
That is (and was) intended. See https://github.com/fmidue/logic-tasks/pull/78#discussion_r1442739002